src/main.js

```diff
@@ -136,85 +130,85 @@ btnBackMain.addEventListener('click', switchPageMain)
 btnShowMain.addEventListener('click', switchPage)
 btnSave.addEventListener('click', savePoster)
 
-
-function savePoster(){
-    posterInstance = new Poster(
-    posterImg.src,
-    posterTitle.innerText,
-    posterQuote.innerText
-  )
-  savedPosters.push(posterInstance)
-};
-
-
 btnShowPoster.addEventListener('click', function() {
-  event.preventDefault()
-  posterTitle.innerText = makeTitle.value
-  posterImg.src = makeImage.value
-  posterQuote.innerText = makeQuote.value
-
-  titles.push(posterTitle.innerText)
-  images.push(posterImg.src)
-  quotes.push(posterQuote.innerText)
-  switchPage()
-
+    event.preventDefault()
+    posterTitle.innerText = makeTitle.value
+    posterImg.src = makeImage.value
+    posterQuote.innerText = makeQuote.value
+
+    titles.push(posterTitle.innerText)
+    images.push(posterImg.src)
+    quotes.push(posterQuote.innerText)
+    switchPage()
 });
 
 // functions and event handlers go here 👇
 // (we've provided one for you to get you started)!
 randoImg = getRandomIndex(images)
 randoTitl = getRandomIndex(titles)
 randoQuote = getRandomIndex(quotes)
+
 posterTitle.innerText = titles[randoTitl]
 posterImg.src = images[randoImg]
 posterQuote.innerText = quotes[randoQuote]
 
+function savePoster() {
+    currentPoster = new Poster(
+    posterImg.src,
+    posterTitle.innerText,
+    posterQuote.innerText)
+
+      if (!savedPosters.includes(currentPoster)) {
+      savedPosters.push(currentPoster)
+      };
```

Comment:
Unfortunately `includes` can't compare full objects as you have it written here. (See these two resources: [Array.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), [String.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) ) 

However every poster instantiation has a unique identifier, so maybe you could look into a way to loop through your saved posters array and see if any of the ids that are already saved match the id of the current poster.